### PR TITLE
Update README.md with threading info

### PR DESCRIPTION
**Pull request by jfunstonRAI**

### CODE REVIEW

**Review by nickrobinson251 — APPROVED** [review]
thanks!

File: README.md [hunk 1]
```diff
@@ -101,13 +101,15 @@ The general recommendation is to treat Rust panics as fatal because Julia tasks
 
 #### Threading Model
 
-Rust object_store uses the [tokio](https://docs.rs/tokio) async runtime.
+Rust object_store uses the [tokio](https://docs.rs/tokio) async runtime. By default tokio sets up a worker thread pool with a number of threads equal to the number of cores.
```
Comment (nickrobinson251) [inline README.md:104]: reference StaticConfig n_threads here?

File: README.md [hunk 2]
```diff
@@ -101,13 +101,15 @@ The general recommendation is to treat Rust panics as fatal because Julia tasks
...
+The unit of scheduling for tokio is a task, and tasks are created by spawn calls. Tasks must be non-blocking and use async/await for I/O operations,
+which also serve as yield points for the cooperative concurrency between tasks. There is work stealing of tasks among the worker thread pools.
 
-TODO
+In object_store_ffi we use buffer_unordered to create a task for each request from Julia (up to a configurable concurrency limit) and allow them to be processed in any order.
```
Comment (nickrobinson251) [inline README.md:108]: reference StaticConfig concurrency_limit here?

Comment (jfunstonRAI) [reply README.md:110]: Good ideas, thanks!

**Pull request merged**